tests/unit/test_dojson_hep_bd0xx.py

```diff
@@ -598,6 +598,66 @@ def test_external_system_numbers_from_035__a_d_h_m_9():
     assert expected == result['035']
 
 
+def test_external_system_numbers_from_035_OSTI_novalue():
+    schema = load_schema('hep')
+    subschema = schema['properties']['external_system_identifiers']
+
+    snippet = (
+        '<datafield tag="035" ind1=" " ind2=" ">'
+        '  <subfield code="9">OSTI</subfield>'
+        '</datafield>'
+    )
+
+    expected = []
+    result = hep.do(create_record(snippet))
+
+    assert validate(result['external_system_identifiers'], subschema) is None
+    assert expected == result['external_system_identifiers']
+
+    expected = []
+    result = hep2marc.do(result)
+
+    assert expected == result.get('035', [])
+
+
+def test_external_system_numbers_from_035_a_z_OSTI():
+    schema = load_schema('hep')
+    subschema = schema['properties']['external_system_identifiers']
+
+    snippet = (
+        '<datafield tag="035" ind1=" " ind2=" ">'
+        '  <subfield code="9">OSTI</subfield>'
+        '  <subfield code="a">892532</subfield>'
+        '</datafield>'
+        '<datafield tag="035" ind1=" " ind2=" ">'
+        '  <subfield code="9">OSTI</subfield>'
+        '  <subfield code="z">897192</subfield>'
+        '</datafield>'
```

Comment:
FYI you need to wrap this in a `<record></record>`, otherwise this is not a valid XML document: they have exactly one root element. `create_record` will ignore the second `datafield`.